spirv_glsl.cpp

```diff
@@ -10020,12 +10020,22 @@ void CompilerGLSL::emit_instruction(const Instruction &instruction)
 		bool pointer = get<SPIRType>(result_type).pointer;
 
 		auto *chain = maybe_get<SPIRAccessChain>(rhs);
+		auto *imgsamp = maybe_get<SPIRCombinedImageSampler>(rhs);
 		if (chain)
 		{
 			// Cannot lower to a SPIRExpression, just copy the object.
 			auto &e = set<SPIRAccessChain>(id, *chain);
 			e.self = id;
 		}
+		else if (imgsamp)
```

Comment:
When does this come up in practice when you say that MSL can hit it, but not GLSL?

Reply:
One of the Vulkan CTS descriptor set indexing tests includes this SPIR-V:

         %31 = OpTypeSampledImage %10
         %32 = OpSampledImage %31 %20 %30
         %33 = OpCopyObject %31 %32


Without this conversion, the `OpCopyObject` processing code here falls into `to_expression()`, which then hits `case TypeCombinedImageSampler:` and throws an error/assertion.

The code comments related to that throw indicate that GLSL does not use `TypeCombinedImageSampler` because `OpSampledImage` converts it to an expression, but MSL certainly does use `TypeCombinedImageSampler`.

Reply:
I see.